paddle/fluid/operators/repeat_interleave_op.h

```diff
@@ -0,0 +1,196 @@
+  paddle::framework::TensorFromVector<RepeatsT>(index_vec, *ctx, index);
+}
+
+template <typename DeviceContext, typename T>
+class RepeatInterleaveKernel : public framework::OpKernel<T> {
+ public:
+  void Compute(const framework::ExecutionContext& context) const override {
+    auto inputs = *context.Input<framework::LoDTensor>("X");
+    auto* output = context.Output<framework::LoDTensor>("Out");
+
+    int dim = context.Attr<int>("dim");
+    if (dim < 0) {
+      dim += inputs.dims().size();
+    }
+
+    int repeats = context.Attr<int>("Repeats");
+    framework::LoDTensor index;
+    if (context.HasInput("RepeatsTensor")) {
+      auto repeats_tensor =
+          context.Input<framework::LoDTensor>("RepeatsTensor");
+
+      PADDLE_ENFORCE_EQ(repeats_tensor->dims()[0] == inputs.dims()[dim], true,
+                        platform::errors::InvalidArgument(
+                            "The length of Input(RepeatsTensor) must be the "
+                            "same as length of Input(X) in axis. "
+                            "But received: [%s], required: [%d].",
+                            repeats_tensor->dims()[0], inputs.dims()[dim]));
+
+      const auto& index_type = repeats_tensor->type();
+      bool index_type_match = index_type == framework::proto::VarType::INT32 ||
+                              index_type == framework::proto::VarType::INT64;
+      PADDLE_ENFORCE_EQ(
+          index_type_match, true,
+          platform::errors::InvalidArgument(
+              "Input(RepeatsTensor) holds the wrong type, it holds %s, but "
+              "desires to be %s or %s",
+              paddle::framework::DataTypeToString(index_type),
+              paddle::framework::DataTypeToString(
+                  framework::proto::VarType::INT32),
+              paddle::framework::DataTypeToString(
+                  framework::proto::VarType::INT64)));
+
+      if (index_type == framework::proto::VarType::INT32) {
+        RepeatsTensor2IndexTensor<DeviceContext, int>(*repeats_tensor, &index);
+        auto output_dim = framework::vectorize(inputs.dims());
+        output_dim[dim] = index.dims()[0];
+        output->Resize(framework::make_ddim(output_dim));
+        IndexSelectInner<DeviceContext, T, int>(context, &inputs, index, output,
+                                                dim);
+      } else if (index_type == framework::proto::VarType::INT64) {
+        RepeatsTensor2IndexTensor<DeviceContext, int64_t>(*repeats_tensor,
+                                                          &index);
+        auto output_dim = framework::vectorize(inputs.dims());
+        output_dim[dim] = index.dims()[0];
+        output->Resize(framework::make_ddim(output_dim));
+        IndexSelectInner<DeviceContext, T, int64_t>(context, &inputs, index,
+                                                    output, dim);
+      }
+    } else if (repeats > 0) {
+      int index_size = inputs.dims()[dim] * repeats;
```

Comment:
inputs.dims()[dim] will return data type of int64_t, index_size should be define as int64_t ?